### ISSUE TITLE
Search API is not notified if we recycle a message

### ISSUE THREAD

**Issue by jdarwood007** [SimpleMachines/SMF#3110]
RemoveTopic.php

```
    // Only remove posts if they're not recycled.
    if (!$recycle)
    {
        // Callback for search APIs to do their thing
        require_once($sourcedir . '/Search.php');
        $searchAPI = findSearchAPI();
        if ($searchAPI->supportsMethod('postRemoved'))
            $searchAPI->postRemoved($message);
```

We only seem to notify the SearchAPI if we are not recycling a post.  This means the API will be out of sync until it manually corrects it.  This does not affect deleting an entire topic.

The way I see it we can fix this 2 ways.
1. We could call postRemoved anyways but with a second parameter called $recycle and let the API handle it if we are recycling a message/topic.
2. We call postModified method and let it treat it as if the post was modified.
3. Setup a new search api method to handle this.

**Comment by Oldiesmann** [comment]
Unless a search API involves storing information about which boards a particular word appears in, this won't be an issue at all (and to be honest I can't think of a reason why we'd need to store that info). The custom search index option stores info about which words are in which posts, so there's no need to update the search index when we recycle a topic.

**Comment by jdarwood007** [comment]
It would be more for returning the proper results by not showing that message if its in a board they can not see.  We do run a method (topicsMoved) when we move a topic entirely.  SMF should still not show them the result because it is in a board they can't see, but the API will be feeding SMF false information.

**Comment by Oldiesmann** [comment]
How is it feeding false info? We build a query to limit the results to those in boards a user can see prior to calling the searchQuery method of the API (see lines 531-585 of Search.php).

**Comment by jdarwood007** [comment]
If you are searching for results on as specific board, that gets passed into the information sent to the Search API.  If that post has moved, the Search API will believe the results are in the old board and doesn't know it now has been moved to a new board that the user doesn't have access to.  The API will feed those matching messages back into SMF.  SMF still corrects this though as it checks to make sure that message is in a board visible to them.

We can close this if it's a non issue.  This is just something I noticed while reviewing the code working on the Sphinx API for 2.1.
